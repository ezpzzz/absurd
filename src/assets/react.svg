<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 100 100">
  <circle cx="50" cy="50" r="10" fill="#61dafb" />
  <ellipse cx="50" cy="50" rx="30" ry="10" stroke="#61dafb" fill="none" />
  <ellipse cx="50" cy="50" rx="10" ry="30" stroke="#61dafb" fill="none" />
</svg>
