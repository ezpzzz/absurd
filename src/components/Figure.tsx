import React from 'react'
import { Bar } from 'react-chartjs-2'
import { Chart as ChartJS, CategoryScale, LinearScale, BarElement, Tooltip, Legend } from 'chart.js'

ChartJS.register(CategoryScale, LinearScale, BarElement, Tooltip, Legend)

interface FigureProps {
  title: string
  data: number[]
}

const Figure: React.FC<FigureProps> = ({ title, data }) => {
  const chartData = {
    labels: data.map((_, idx) => `Item ${idx + 1}`),
    datasets: [
      {
        label: title,
        data,
        backgroundColor: 'rgba(75, 192, 192, 0.5)',
      },
    ],
  }

  return (
    <figure>
      <Bar data={chartData} aria-label={title} role="img" />
      <figcaption>{title}</figcaption>
      <table>
        <thead>
          <tr>
            <th>Label</th>
            <th>Value</th>
          </tr>
        </thead>
        <tbody>
          {data.map((value, idx) => (
            <tr key={idx}>
              <td>{`Item ${idx + 1}`}</td>
              <td>{value}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </figure>
  )
}

export default Figure
