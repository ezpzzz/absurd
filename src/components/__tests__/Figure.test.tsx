import React from 'react'
import { render, screen } from '@testing-library/react'
import { vi } from 'vitest'
import Figure from '../Figure'

vi.mock('react-chartjs-2', () => ({
  Bar: (props: any) => <canvas {...props} />,
}), { virtual: true })

vi.mock('chart.js', () => ({
  Chart: { register: () => {} },
  CategoryScale: {},
  LinearScale: {},
  BarElement: {},
  Tooltip: {},
  Legend: {},
}), { virtual: true })

describe('Figure', () => {
  it('renders chart with alt text and table snapshot', () => {
    const { container } = render(<Figure title="Test Chart" data={[1, 2, 3]} />)
    screen.getByRole('img', { name: 'Test Chart' })
    screen.getByText('Test Chart')
    expect(container).toMatchSnapshot()
  })
})
