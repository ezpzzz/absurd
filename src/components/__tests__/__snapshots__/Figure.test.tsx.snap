// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Figure > renders chart with alt text and table snapshot 1`] = `
<div>
  <figure>
    <canvas
      aria-label="Test Chart"
      data="[object Object]"
      role="img"
    />
    <figcaption>
      Test Chart
    </figcaption>
    <table>
      <thead>
        <tr>
          <th>
            Label
          </th>
          <th>
            Value
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            Item 1
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <td>
            Item 2
          </td>
          <td>
            2
          </td>
        </tr>
        <tr>
          <td>
            Item 3
          </td>
          <td>
            3
          </td>
        </tr>
      </tbody>
    </table>
  </figure>
</div>
`;
